src/components/Camera/withFailureHandling.js

```diff
@@ -0,0 +1,51 @@
+// @flow
+import * as React from 'react'
+import { h, Component } from 'preact'
+import CameraError from '../CameraError'
+
+const generalError = { name: 'CAMERA_NOT_WORKING', type: 'error' }
+
+type State = {
+  hasError: boolean,
+}
+
+type Props = {
+  onError: ?string => void,
+}
+
+export default <WrappedProps: *>(
```

Comment:
Is this a flow syntax?

Reply:
Yes! https://flow.org/en/docs/react/hoc/#toc-injecting-props-with-a-higher-order-component